src/components/templates/PartyMap/PartyMap.tsx

```diff
@@ -32,59 +27,20 @@ export const PartyMap: React.FC = () => {
   const { recentVenueUsers } = useRecentVenueUsers();
 
   const currentVenue = useSelector(partyMapVenueSelector);
-  const venues = useSelector(orderedVenuesSelector);
 
-  const [isRoomModalOpen, setRoomModalOpen] = useState(false);
   const [selectedRoom, setSelectedRoom] = useState<Room | undefined>();
 
   const selectRoom = useCallback((room: Room) => {
     setSelectedRoom(room);
-    setRoomModalOpen(true);
   }, []);
 
-  // Note: we explicitly don't unset selectedRoom here as that would cause RoomModal to close abruptly
   const unselectRoom = useCallback(() => {
-    setRoomModalOpen(false);
+    setSelectedRoom(undefined);
   }, []);
 
-  // TODO: extract this into a reusable hook/similar
-  const enterRoom = useCallback(
-    (room: Room) => {
-      if (!room || !user) return;
-
-      // TODO: we could process this once to make it look uppable directly? What does the data key of venues look like?
-      const roomVenue = venues?.find((venue) =>
-        room.url.endsWith(`/${venue.id}`)
-      );
-
-      const nowInUTCSeconds = getCurrentTimeInMilliseconds();
-
-      const roomName = {
-        [`${currentVenue.name}/${room.title}`]: nowInUTCSeconds,
-        ...(roomVenue ? { [currentVenue.name]: nowInUTCSeconds } : {}),
-      };
-
-      trackRoomEntered(user, roomName, profile?.lastSeenIn);
-
-      if (roomVenue) {
-        openUrl(venueInsideUrl(roomVenue.id));
-        return;
-      }
-      openRoomUrl(room.url);
-    },
-    [profile, user, currentVenue, venues]
-  );
-
-  // Note: since we explicitly don't unset selectedRoom in unselectRoom, we need to check if
-  //   the RoomModal is open to know if we have a room selected
-  const enterSelectedRoom = useCallback(() => {
-    if (!selectedRoom || !isRoomModalOpen) return;
-
-    enterRoom(selectedRoom);
-  }, [enterRoom, isRoomModalOpen, selectedRoom]);
-
   // Find current room from url
   const { roomTitle } = useParams();
+
   const currentRoom = useMemo(() => {
```

Comment:
This seems similar if not exactly the same as the code in `useRoom`, so this can probably be refactored to use that.

Reply:
How is it the same?

Reply:
Looking at the full amount of it, perhaps it isn't in its entirety. I believe the part I was referring to was finding the room (which again, may not be the same, but while skimming and reviewing the code, it seemed similar):

```
currentVenue.rooms.find(
      (venueRoom) =>
        createUrlSafeName(venueRoom.title) === createUrlSafeName(roomTitle)
    );
```